registry/client/transport/http_reader.go

```diff
@@ -190,30 +190,36 @@ func (hrs *httpReadSeeker) reader() (io.Reader, error) {
 	if resp.StatusCode >= 200 && resp.StatusCode <= 399 {
 		if hrs.readerOffset > 0 {
 			if resp.StatusCode != http.StatusPartialContent {
+				resp.Close()
```

Comment:
At least these should be `resp.Body.Close()`, because `resp.Close` is a boolean (not a func).